akka-docs/src/main/paradox/cluster-usage.md

```diff
@@ -1013,3 +1013,20 @@ Related config properties: `akka.cluster.use-dispatcher = akka.cluster.cluster-d
 Corresponding default values: `akka.cluster.use-dispatcher =`.
 
 @@@
+
+### Configuration Compatibility Check
+
+Creating a cluster is about deploying two or more nodes and make then behave as if they were one single application. Therefore it's extremely important that all nodes in a cluster are configured with the same settings. 
+
+The Configuration Compatibility Check feature ensures that all nodes in a cluster have a compatible configuration. Whenever a new node is joining an existing cluster, a subset of its configuration settings (only those that are required to be checked) is sent to the nodes in the cluster for verification. Once the configuration is checked on the cluster side, the cluster sends back its own set of required configuration settings. The joining node will them verify if it's compliant with the cluster configuration. The joining node will only proceed if all checks passes, on both sides.   
+
+
+@@@ note
+
+Configuration Compatibility Check is enabled by default, but can be disabled by setting `akka.cluster.configuration-compatibility-check.enforce-on-join = off`. This is specially useful when performing rolling updates. Obviously this should only be done if a complete cluster shutdown isn't an option. A cluster with nodes with different configuration settings may lead to data loss or data corruption. 
+
+This setting should only be disabled on the joining nodes. The checks are always performed on both sides, and warnings are logged. In case of incompatibilities, it's the joining node to decide if the process should be interrupted or not.  
+
+If you are performing a rolling update on cluster using Akka 2.5.9 or prior (thus, not supporting this feature), you will need to set `akka.cluster.configuration-compatibility-check.enforce-on-join = off`, otherwise the new nodes will refuse to join a cluster that is incapable of performing the necessary checks.  
```

Comment:
Here I'm referring to Akka 2.5.9. Should be changed if this PR is not merged in 2.5.10

Reply:
Wonder if this is the right behavior when joining 2.5.10 to 2.5.9? Could we allow that joining?

Reply:
By default, we won't allow it, but we must leave a backdoor in case a complete cluster shutdown is not acceptable.

Reply:
As discussed offline: it sounds like we could detect when joining a pre-2.5.10 node and in that case make the validation pass even though the 2.5.9 node did not respond with any configuration and `enforce-on-join = on`

Reply:
Sure. I did it, but it's missing a note on documentation stating that. I will do it.